src/main/java/net/snowflake/ingest/utils/ParameterProvider.java

```diff
@@ -384,15 +450,17 @@ public long getMaxAllowedRowSizeInBytes() {
     return (val instanceof String) ? Long.parseLong(val.toString()) : (long) val;
   }
 
-  /**
-   * @return The max number of chunks that can be put into a single BDEC or blob registration
-   *     request.
-   */
-  public int getMaxChunksInBlobAndRegistrationRequest() {
+  /** @return The max number of chunks that can be put into a single BDEC. */
+  public int getMaxChunksInBlob() {
+    Object val = this.parameterMap.getOrDefault(MAX_CHUNKS_IN_BLOB, MAX_CHUNKS_IN_BLOB_DEFAULT);
```

Comment:
we should use the iceberg_default here to make this accurate?

Reply:
The [setParameterMap()](https://github.com/snowflakedb/snowflake-ingest-java/blob/3d95d22ea67c87a5d758d5aacd1f497a025302a7/src/main/java/net/snowflake/ingest/utils/ParameterProvider.java#L239) and [checkAndUpdate()](https://github.com/snowflakedb/snowflake-ingest-java/blob/3d95d22ea67c87a5d758d5aacd1f497a025302a7/src/main/java/net/snowflake/ingest/utils/ParameterProvider.java#L122) will guarantee that the Iceberg default value is always in the map.

Reply:
In that case lets remove the getOrDefault and throw if get returns null?

Reply:
@sfc-gh-tzhang any preferences on this?

Reply:
Discussed offline with Alec, I think we should keep the old behavior and not adding everything in the map, otherwise we don't know if it's actually an override or not as all these values will be in the map.

Reply:
sounds good! I mainly want to ensure there's precisely one way to do defaulting, either by adding everything to the map or at read time in this get method. 

Also the new enforceDefault block won't have to change even if we don't add the default value to the map.

Reply:
Changed the behavior to the following